### DISCUSSION TITLE
Can NNI be used in TF Estimator?

### DISCUSSION THREAD

**Discussion by fireindark707** [microsoft/nni#3751] [Q&A] [answered]
**Environment**:
- NNI version: latest
- NNI mode (local|remote|pai): local
- Client OS: Ubuntu
- Server OS (for remote mode only):
- Python version: 3.7
- PyTorch/TensorFlow version: TensorFlow-gpu 1.14.0
- Is conda/virtualenv/venv used?: yes
- Is running in Docker?: no

**Log message**:
 - nnimanager.log: 
 - dispatcher.log:
 - nnictl stdout and stderr:
 
I want to add the intermediate result to the Tensorflow training, but since the code is executed in estimator.train, I don't know how to get the intermediate result. 

`logging_hook = tf.train.LoggingTensorHook({"loss": total_loss}, every_n_iter=10)`

I see that the code has a loss output like the above, but I don't know how to make nni get the loss

**Comment by ultmaster** [comment, marked as answer]
`SessionRunHook` can be inherited and you can implement some hook like `LoggingTensorHook` and put `nni.report_intermediate_result` into it. You can refer to the source code of `LoggingTensorHook` for reference.
